engines/python/setup/djl_python/xgboost_handler.py

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env python
+#
+# Copyright 2025 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file
+# except in compliance with the License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS"
+# BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, express or implied. See the License for
+# the specific language governing permissions and limitations under the License.
+
+import numpy as np
+import os
+import pickle as pkl
+from io import StringIO
+from typing import Optional
+from djl_python import Input, Output
+from djl_python.encode_decode import decode
+from djl_python.utils import find_model_file
+from djl_python.service_loader import get_annotated_function
+from djl_python.import_utils import xgboost as xgb
+
+
+class XGBoostHandler:
+
+    def __init__(self):
+        self.model = None
+        self.initialized = False
+
+    def initialize(self, properties: dict):
+        model_dir = properties.get("model_dir")
+
+        extensions = ["pkl", "model", "xgb", "bst", "json", "ubj"]
+
+        model_file = find_model_file(model_dir, extensions)
+        if not model_file:
+            raise FileNotFoundError(
+                f"No XGBoost model file found in {model_dir}")
+
+        try:
+            with open(model_file, 'rb') as f:
+                self.model = pkl.load(f)
```

Comment:
Similar to skl, we should categorize the file formats as safe and unsafe, create the appropriate serving properties. We should always prefer the safe formats - json and ubjson unless the user tells us otherwise.